compiler/next/lib/parsing/Parser.cpp

```diff
@@ -71,7 +71,9 @@ static void updateParseResult(ParserContext* parserContext) {
   for (ParserError & parserError : parserContext->errors) {
     // Need to convert the error to a regular ErrorMessage
     Location loc = parserContext->convertLocation(parserError.location);
-    builder->addError(ErrorMessage(ID(), loc, parserError.message));
+    auto errMsg = ErrorMessage(ID(), loc, parserError.message,
+                               parserError.kind);
```

Comment:
Huh, I guess not all parsing errors are kind SYNTAX?

Reply:
Yeah there are errors that are more to do with "this expression doesn't make sense in this particular context" that I don't necessarily think we'd want to treat as a syntax error. The old compiler emits a lot of these in `checkParsed`, and I think we might want to do something similar before/after adding IDs, but they would use `noteError` regardless?

Reply:
I think doing some of these during parsing is just fine and it doesn't really bother me if it happens before / after adding IDs.